src/NuGet.Core/NuGet.Packaging.Core/IPackageCoreWriter.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Packaging.Core
+{
+    /// <summary>
+    /// Allows package modifications.
+    /// </summary>
+    public interface IPackageCoreWriter
+    {
+        /// <summary>
+        /// Remove if the path exists.
+        /// </summary>
+        /// <param name="path">Relative file path in package.</param>
+        /// <returns>True if the file existed before the remove.</returns>
+        /// <param name="token">CancellationToken</param>
+        Task<bool> RemoveAsync(string path, CancellationToken token);
```

Comment:
Unless the return value is actually useful, you could simply return `Task` like [here](https://github.com/NuGet/NuGet.Client/pull/1728/files#diff-85475b61cd04990ad26491dae1c1ba7fR57).